modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb

```diff
@@ -36,10 +36,42 @@ def submit
         render json: { error_message: "Error: #{e.message}" }, status: :internal_server_error
       end
 
+      # Modified from claim_documents_controller.rb:
+      def unlock_file(file, file_password)
+        return file unless File.extname(file) == '.pdf' && file_password
+
+        pdftk = PdfForms.new(Settings.binaries.pdftk)
+        tmpf = Tempfile.new(['decrypted_form_attachment', '.pdf'])
+
+        begin
+          pdftk.call_pdftk(file.tempfile.path, 'input_pw', file_password, 'output', tmpf.path)
+        rescue PdfForms::PdftkError => e
+          file_regex = %r{/(?:\w+/)*[\w-]+\.pdf\b}
+          password_regex = /(input_pw).*?(output)/
+          sanitized_message = e.message.gsub(file_regex, '[FILTERED FILENAME]').gsub(password_regex, '\1 [FILTERED] \2')
+          log_message_to_sentry(sanitized_message, 'warn')
+          raise Common::Exceptions::UnprocessableEntity.new(
+            detail: I18n.t('errors.messages.uploads.pdf.incorrect_password'),
+            source: 'IvcChampva::V1::UploadsController'
+          )
+        end
+
+        file.tempfile.unlink
+        file.tempfile = tmpf
+        file
+      end
+
```

Comment:
This is the logic we were missing - now PDFs can actually be unlocked when the user provides a password on submission.